### PR TITLE
Integration with Peer Sharing

### PR DESCRIPTION
**Pull request by bolt12**
Integration with https://github.com/input-output-hk/ouroboros-network/pull/4019

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -89,3 +89,14 @@ package snap-server
...
+  type: git
+  location: https://github.com/input-output-hk/ouroboros-network
+  tag: e4a222f8e0f5d55591cc14b062332a8a61e85724
+  --sha256: sha256-LrrtA5KR1mFgSQmUZUWIXAS23bUbEVpnnwH1ehH2oWM=
+  subdir: ouroboros-network
+          ouroboros-network-api
+          ouroboros-network-protocols
+          ouroboros-network-framework
+          ouroboros-consensus-diffusion
+          ouroboros-consensus-byron
```
Comment (coot) [inline cabal.project:102]: This `s-r-p` will be removed once we publish new versions to `CHaP`.

**Comment by coot** [comment]
I dropped the `stylish-haskell` changes; this CI check is not enforced.

**Review by deepfire — APPROVED** [review]
Thank you @bolt12 !

**Review by angerman — APPROVED** [review]
Going to trust @coot and @deepfire with their approval here. Nothing particularly sticks out to me from the review. There's one HLint complaint, but we should probably address _all_ hlint complains.

**Review by erikd — APPROVED** [review]
Seems sane.

**Comment by angerman** [comment]
This needs This needs #5160 to unbreak master first.

**Comment by angerman** [comment]
#5160 was just merged 🥳

**Comment by coot** [comment]
I addressed 2/3 hlint complains; I left one for two reasons: I prefer explicit arguments over eta reduction, and it wasn't introduced in this PR.

**Comment by coot** [comment]
I also rebased the branch, everything still compiles locally.

**Comment by bolt12** [comment]
Thank you for addressing this while I was out @coot ! Hurray!

**Pull request merged**